gem-changelog-update-check/action.yml

```diff
@@ -0,0 +1,30 @@
+name: Check if CHANGELOG needs updating
+description: Checks if files in lib/ have changed; if so, check if CHANGELOG.md has been updated
+author: Workflow Improvement Team
+inputs:
+  changelog-file-path:
+    description: Relative path to the CHANGELOG file
+    required: false
+    default: CHANGELOG.md
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Check if CHANGELOG needs updating
+      run: |
+        file_regex='^lib/|^bin/|^spec/|\.gemspec$'
+        changes="$(git diff --name-only --diff-filter=ACMRT origin/main HEAD)"
```

Comment:
`--diff-filter` should probably include **D**eleted too; if you delete something from `lib` there should be an explanation in the changelog, even if it's a refactor.